File: src/decoders/new-discussion-recipient-sub-subject.ts
import type { NewDiscussionRecipientSubSubject } from "~/models";

export const decodeNewDiscussionRecipientSubSubject = (sub: any): NewDiscussionRecipientSubSubject => {
  return {
    id: sub.N,
    name: sub.L,
    from: sub.libelleMatiere
  };
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/decoders/new-discussion-recipient-sub-subject.kt
package ink.literate.pawnote.decoders

import ink.literate.pawnote.models.NewDiscussionRecipientSubSubject
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonPrimitive

fun decodeNewDiscussionRecipientSubSubject (sub: JsonObject) = NewDiscussionRecipientSubSubject(
    id = sub["N"]!!.jsonPrimitive.content,
    name = sub["L"]!!.jsonPrimitive.content,
    from = sub["libelleMatiere"]!!.jsonPrimitive.content
)